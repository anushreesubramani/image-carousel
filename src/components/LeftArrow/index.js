import React from "react";
import PropTypes from "prop-types";
import { LeftArrowWrapper } from "./styles";
import arrow from "assets/arrow.svg";

const LeftArrow = props => {
  return (
    <LeftArrowWrapper data-testid="leftclick" onClick={props.goToPrevSlide}>
      <img src={arrow} alt="leftarrow" />
    </LeftArrowWrapper>
  );
};

LeftArrow.propTypes = {
  goToPrevSlide: PropTypes.func
};

export default LeftArrow;
