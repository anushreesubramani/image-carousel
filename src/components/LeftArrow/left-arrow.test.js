import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import LeftArrow from "components/LeftArrow";

afterEach(cleanup);

describe("testing left arrorw", () => {
  it("rendering left arrow and click", () => {
    const handleClick = jest.fn();
    const { getByAltText, getByTestId, container } = render(
      <LeftArrow goToPrevSlide={handleClick} />
    );
    const arrowElement = getByTestId("leftclick");
    fireEvent.click(arrowElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(getByAltText("leftarrow").alt).toBe("leftarrow");
    expect(container).toMatchSnapshot();
  });
});
