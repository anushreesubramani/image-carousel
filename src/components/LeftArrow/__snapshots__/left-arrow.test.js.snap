// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing left arrorw rendering left arrow and click 1`] = `
<div>
  <div
    class="sc-bwzfXH gpDCHp sc-bdVaJa ePgJGs"
    data-testid="leftclick"
  >
    <img
      src="arrow.svg"
      alt="leftarrow"
    />
  </div>
</div>
`;
